src/Chapter2.hs

```diff
@@ -727,8 +739,11 @@ value of the element itself
 
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
+
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate xs = foldl (++) [] (map (\x -> replicate x x) xs)
```

Comment:
Nice solution!

You correctly noticed that after using `map` you will get a list of lists. To flatten such a list, you can use the `concat` function instead of `foldl`:

```haskell
ghci> concat [[3, 1, 2], [4, 0, 1]]
[3,1,2,4,0,1]
```

This pattern of using `map` and `concat` is very common, so the standard library has even a standard function `concatMap` which is a combination of `map` and `concat` (as the name suggests).